src/integration-test/java/io/aiven/kafka/connect/gcs/AvroParquetIntegrationTest.java

```diff
@@ -0,0 +1,368 @@
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+
+        // TODO more robust way to detect that Connect finished processing
+        Thread.sleep(OFFSET_FLUSH_INTERVAL_MS * 2);
+
+        final List<String> expectedBlobs = List.of(
+                getBlobName(0, 0, compression),
+                getBlobName(1, 0, compression),
+                getBlobName(2, 0, compression),
+                getBlobName(3, 0, compression));
+        assertIterableEquals(expectedBlobs, testBucketAccessor.getBlobNames(gcsPrefix));
+
+        final Map<String, List<GenericRecord>> blobContents = new HashMap<>();
+        for (final String blobName : expectedBlobs) {
+            final var records =
+                    ParquetUtils.readRecords(
+                            tmpDir.resolve(Paths.get(blobName)),
+                            testBucketAccessor.readBytes(blobName)
+                    );
+            blobContents.put(blobName, records);
+        }
+        cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final var name = "user-" + cnt;
+                final var value = "value-" + cnt;
+                final String blobName = getBlobName(partition, 0, "none");
+                final GenericRecord record = blobContents.get(blobName).get(i);
+                final var expectedKey = "key-" + cnt;
+                final var expectedValue = "{\"name\": \"" + name + "\", \"value\": \"" + value + "\"}";
+                assertEquals(expectedKey, record.get("key").toString());
+                assertEquals(expectedValue, record.get("value").toString());
+                assertNotNull(record.get("offset"));
+                assertNotNull(record.get("timestamp"));
+                assertNull(record.get("headers"));
+                cnt += 1;
+            }
+        }
+    }
+
+    @Test
+    final void valueComplexType(@TempDir final Path tmpDir)
+            throws ExecutionException, InterruptedException, IOException {
+        final Map<String, String> connectorConfig = basicConnectorConfig();
+        final String compression = "none";
+        connectorConfig.put("format.output.fields", "value");
+        connectorConfig.put("format.output.fields.value.encoding", "none");
+        connectorConfig.put("file.compression.type", compression);
+        connectorConfig.put("format.output.type", "parquet");
+        connectRunner.createConnector(connectorConfig);
+
+        final Schema valueSchema =
+                SchemaBuilder.record("value")
+                        .fields()
+                        .name("name").type().stringType().noDefault()
+                        .name("value").type().stringType().noDefault()
+                        .endRecord();
+
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        int cnt = 0;
+        for (int i = 0; i < 10; i++) {
```

Comment:
I think there should be a generator for such a records somewhere

Reply:
true, but so far it is not clear. the problem is that we have same tests for S3 :-(

Reply:
I personally think that we need merge `aiven-kafka-commons`, `gcs` and `s3` connectors into one repo.